src/CrashlogFile.ts

```diff
@@ -125,4 +219,192 @@ export class CrashlogFile {
+        } else {
+            parsedStackTraceSection.errorMessage = firstLine;
+            lines.shift();
+        }
+
+        const stackTraceSections: Array<{ sectionType: StackTraceSectionType; lines: string[] }> = [];
+
+        let currentSection: StackTraceSectionType | undefined;
+        let foundSectionHeader = false;
+
+        // Separate the block in two sections: Backtrace and LocalVariables
+        for (const line of lines) {
+            if (line === 'Local Variables:') {
+                currentSection = StackTraceSectionType.LocalVariables;
+                foundSectionHeader = true;
+            } else if (line === 'Backtrace:') {
+                currentSection = StackTraceSectionType.BackTrace;
+                foundSectionHeader = true;
+            } else {
+                if (foundSectionHeader && currentSection) {
+                    let sectionIndex = stackTraceSections.findIndex(x => x.sectionType === currentSection);
+                    if (sectionIndex === -1) {
+                        stackTraceSections.push({
+                            sectionType: currentSection,
+                            lines: [line]
+                        });
+                    } else {
+                        stackTraceSections[sectionIndex].lines.push(line);
+                    }
+                }
+            }
+        }
+
+        // Process each section
+        for (const stackTraceSection of stackTraceSections) {
+            switch (stackTraceSection.sectionType) {
+                case StackTraceSectionType.LocalVariables:
+                    parsedStackTraceSection.localVariables = this.parseStackTraceLocalVariables(stackTraceSection.lines);
+                    break;
+                case StackTraceSectionType.BackTrace:
+                    parsedStackTraceSection.stackFrame = this.parseStackFrames(stackTraceSection.lines);
+                    break;
+            }
+        }
+
+        return parsedStackTraceSection;
+    }
+
+    /**
+     * Parses the stack frames.
+     * Extracts the scope and location of each stack frame.
+    */
+    public parseStackFrames(lines: string[]): StackFrame[] {
+        const stackTrace: StackFrame[] = [];
+
+        let stackFrame: StackFrame = { scope: '', reference: undefined };
+
+        for (const line of lines) {
+            if (line === '') {
+                continue;
+            }
+            if (/#[0-9]+\s+./.exec(line)) {
+                const [_, ...scopeAsArray] = line.split(/\s+/);
+                stackFrame.scope = scopeAsArray.join(' ').trim();
+            } else if (/file\/line:\s+./.exec(line)) {
+                const [_, ...pkgLocationAsArray] = line.split(/\s+/);
+
+                const match = /(\w+:\/.*?)\((\d+)\)/.exec(pkgLocationAsArray.join(' ').trim());
+                if (match) {
+                    // We need the range to calculate the reference index. To generate the range, we need the line number.
```

Comment:
@TwitchBronBron I honestly couldn't think of a more elegant solution for this. I wanted to make the subparsers for each crash section agnostic to the order of the sections, so Roku can miss sending, for example, the hardware information and the parser could still figure out the stack trace. This resulted in the subparser for the stack trace not having the original line number for each of its section lines. Hence this code...